### DISCUSSION TITLE
[Help] 请教个问题，我想微调chatglm以更好的完成text2sql的场景，请问prompt该怎么设计比较好?

### DISCUSSION THREAD

**Discussion by goingHan** [zai-org/ChatGLM-6B#1380] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

我目前使用的prompt的格式类似这样的:

`
_我们要将用户的问题翻译为一个mysql的sql语句。
问题涉及的数据表的信息是：

    表名：a_activity_instance
    表结构：
    activity_instance_id    活动实例ID
    activity_type_id    活动类型ID
    activity_code   活动编码
    busi_category   业务类型，参见"活动业务分类"表的配置

用户的问题是：

    get all activity_instance_id of activity?

将用户的问题翻译为mysql的sql语句，sql语句是:

`

想请教一下这样正确不？ 如果这样的话，pre_seq_len 就要设置的比较大，会不会对最后的效果有影响？ 麻烦大佬给一些建议，跪谢。

### Expected Behavior

None

### Steps To Reproduce

None

### Environment

```markdown
- OS:
- Python:
- Transformers:
- PyTorch:
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :
```

**Comment by liaoxingjian** [comment]
我也有这个问题🥹

**Comment by nuo-o** [comment]
我也有类似的问题，我现在是很笨的方法。先建立本地的知识库，写明每个表能干什么，再用langchain+chatglm 回答。很笨。因为有一大堆需要配置的。

**Comment by PerseveranceZ** [comment]
+1

**Comment by nuo-o** [comment]
看看这个：https://github.com/yysirs/ChatSQL

**Comment by stormeyes** [comment]
> 看看这个：https://github.com/yysirs/ChatSQL

这个是基于向量搜索的，不是基于微调的

**Comment by kanseaveg** [comment]
正在尝试，敬请期待~
